docs/integration-testing.md

```diff
@@ -19,53 +18,66 @@ typically run faster. Sometimes that's not (easily) possible, especially when an
 
 Therefore, in many cases writing unit tests is more involved that writing an integration test, for example say we wanted
-to test our implementation of a CosmosDB-backed queue. We would have to mock the behaviour of the CosmosDB API, which -
+to test our implementation of a CosmosDB-backed queue. We would have to mock the behavior of the CosmosDB API, which -
 while certainly possible - can get complicated pretty quickly. Now we still might do that for simpler scenarios, but
 eventually we might want to write an integration test that uses a CosmosDB test instance.
 
 ## Coding Guidelines
 
-An integration test is annotated with `@IntegrationTest`, which causes the test runner to ignore it unless the
-`RUN_INTEGRATION_TEST` environment variable is set to `true`.
+EDC codebase has few annotations to help you write and categorize integration tests. The following are some of the available ones:
 
-All integration tests should have the `"...IntegrationTest"` postfix to distinguish them clearly from unit tests. They
-should reside in the same package as unit tests because all tests should maintain package consistency to their test
-subject.
+- `@IntegrationTest`: Marks an integration test with `integration-test` Junit tag. This is the default tag and can be used if you do not want to specify any other tags on your test to do further categorization.
 
-Any credentials, secrets, passwords, etc. that are required by the integration tests should be passed in using
-environment variables. A good way to access them is `ConfigurationFunctions.propOrEnv()` because then the credentials
-can also be supplied via system properties.
+Below annotations are used to categorize integration tests based on their technologies. All of these annotations are composite annotations and contains `@IntegrationTest` annotation as well.
 
-There is no one-size-fits-all guideline whether to perform setup tasks in the `@BeforeAll` or `@BeforeEach`, it will
-depend on the concrete system you're using. As a general rule of thumb long-running one-time setup should be done in
-the `@BeforeAll` so as not to extend the run-time of the test unnecessarily. In contrast, in most cases it is **not**
-advisable to deploy/provision the external system itself in either of those methods. In other words, provisioning a
-CosmosDB or spinning up a Postgres docker container directly from test code should generally be avoided, because it will
-introduce code that has nothing to do with the test and may cause security problems (privilege escalation through the
-Docker API), etc.
+- `@AzureStorageIntegrationTest`: Marks an integration test with `azure-storage-integration-test` Junit tag. This should be used when the integration test is related to `Azure Storage`.
+- `@AzureCosmosDbIntegrationTest`: Marks an integration test with `azure-cosmosdb-integration-test` Junit tag. This should be used when the integration test is related to `Azure CosmosDB`.
```

Comment:
nit: why are the tags not named exactly like the annotations? Wouldn't it be simpler to understand?

Reply:
Good suggestion. Update tag names and docs.